### DISCUSSION TITLE
Link transactions from downstream jobs to the parentstream one

### DISCUSSION THREAD

**Discussion by v1v** [jenkinsci/opentelemetry-plugin#98] [Ideas]
### What

Given the [`build` step](https://plugins.jenkins.io/pipeline-build-step/) that triggers a downstream job, should the transaction between the parentstream and downstream jobs being linked somehow?

### Why

Facilitate the navigation between parentstream and downstream distributed traces.

### Example

```groovy
pipeline {
  agent any
  stages {
    stage('build') {
      ...
    }
    stage('downstream') {
      steps {
        build(job: 'foo', parameters: [credentials('parameter-name', 'credentials-id')])
      }
    }
  }
}
```

There will be two transactions:
1) the one for the pipeline
2) the one for the foo job.

### Caveats

Build step allows to run the build with the `propagate` flag 

> If enabled (default state), then the result of this step is that of the downstream build (e.g., success, unstable, failure, not built, or aborted). If disabled, then this step succeeds even if the downstream build is unstable, failed, etc.; use the result property of the return value as needed.

**Comment by cyrille-leclerc** [comment]
I love this idea to connect the 2 traces making it a big distributed trace.
I'm not sure to catch the challenge when [`propagate`](https://www.jenkins.io/doc/pipeline/steps/pipeline-build-step/) is set to `true` (eg `build (job: 'my-downstream-pipeline', propagate:  true)`) as the trace is just a visualization of what the pipeline does, the trace doesn't change the behaviour of the pipeline.

**Comment by cyrille-leclerc** [comment]
FYI I started looking at the implementation to connect 2 jobs together. There is a bit of work, I'm on it.

**Comment by cyrille-leclerc** [comment]
See
* https://github.com/jenkinsci/opentelemetry-plugin/pull/208
